### PR TITLE
pretrain composition model

### PR DESCRIPTION
**Pull request by Blubberli**
This PR includes the transweigh composition model that returns a composed phrase. A loss function has been added that computes the cosine distance between two given embeddings. Thus a composition model can be pretrained on a general task, e.g. reconstruct some gold standard vectors. A test has been added as well. The dataset and the pretraining method will be added an a separate PR, otherwise it would be too chaotic.

### CODE REVIEW

**Review by evvahu — APPROVED** [review]
nice!
Do we need a training script for this?

**Comment by Blubberli** [comment]
I added a pertaining scripts. can you check that?
